### ISSUE TITLE
Calling csound via exec() in PHP hangs Apache

### ISSUE THREAD

**Issue by belbernard** [csound/csound#1633]
Hello!
On my application (PHP in MAMP environment on a Mac) calling csound via exec() or system() commands is working perfectly with version 6.15 of Csound.
For instance:

        $csound_path = /usr/local/bin/";
	$command = $csound_path."csound --version";
	system($command,$return_var);

However, after installing version 6.17, the PHP page no longer loads…

In the Mac Terminal, command " /usr/local/bin/csound --version" still works perfectly with both versions 6.15 and  6.17
Is there anything wrong in the configuration?

Thanks in advance!

**Comment by belbernard** [comment]
You can check for instance the following PHP file with a MAMP server:

<?php
$csound_path = "/usr/local/bin/";
$command = $csound_path."csound --version";
echo $command."<br />";
system($command,$return_var);
echo "OK";
?>

It works with Csound 6.15, but it hangs with 6.17.

**Comment by belbernard** [comment]
Apache error log says:
csound command: Segmentation fault

**Comment by belbernard** [comment]
Note that commands produced by my application (Bol Processor) work with Csound 6.17 if they are sent via the Unix terminal. For instance:

/usr/local/bin/csound --wave -o ../my_output/Mozart@60676.wav ../csound_resources/0-default.orc ../my_output/Mozart.sco

Yest the same command will produce a segmentation fault if sent via exec() or system() via Apache (in MAMP).

**Comment by vlazzarini** [comment]
I have installed Apache, PHP and now MAMP but I have no idea how to test this. I tried to open the php file on chrome and the only thing that happens is that I see the file contents.

**Comment by belbernard** [comment]
Installing MAMP created a folder named "htdocs":

Applications/MAMP/htdocs/ (on a Mac)

Your PHP file should be placed in that folder. Suppose it is called "test.php".

Then, while MAMP is running Apache, follow this URL:

http://localhost/test.php

(Note that, by default, you won't use "https".)

**Comment by vlazzarini** [comment]
That's what I said I did. I have tried both with just apache, then I installed MAMP and the same thing results, which is just the browser showing the text contents of the file.

**Comment by vlazzarini** [comment]
ok, I just tried wrapping the PHP code in a tag

```
<?php   
$csound_path = "/usr/local/bin/";
$command = $csound_path."csound --version";
system($command,$return_var);
echo $return_var;
?>
```
and I get 0 on the browser screen. So I suppose this is running as it should?

**Comment by vlazzarini** [comment]
I also tried

```
<?php   
$csound_path = "/usr/local/bin/";
$command = $csound_path."csound trapped.csd -odac";
system($command,$return_var);
echo $return_var;
?>
```
and I can hear the piece played to DAC.

**Comment by vlazzarini** [comment]
Ok, I did a bit more digging, and with MAMP Pro 6.6 it is crashing alright. However with the apache server and PhP I built and installed here, it's running as reported above. So I don't know, it's something with MAMP perhaps?

**Comment by belbernard** [comment]
It would be difficult to invite users of Bol Processor BP3 (https://bolprocessor.org/check-bp3) to install Apache by hand rather than using a "ready-made" solution such as MAMP.

On the documentation I recommended  using Csound 6.15 until we've sold the "segmentation fault" problem. We're not yet using advanced features of Csound anyway… It would help us tremendously that you check MAMP Pro 6.6 with version 6.15 of Csound which you can download from this page: https://github.com/csound/csound/releases/tag/6.15.0

(On Mac it is easy to switch versions, just reinstalling the package in a single click…)

**Comment by vlazzarini** [comment]
yes, if 6.15 works keep it to that for the moment.
I don't really know what is causing the issue, whether it is something in Csound. I was able to add some tracing and locate the issue within the plugin module loading stage.

**Comment by belbernard** [comment]
"yes, if 6.15 works keep it to that for the moment"
It works with MAMP Pro version 5.7 but I haven't checked MAMP Pro version 6.6 (and MacOS 12.3) which you have…

**Comment by belbernard** [comment]
"I have fixed this in the latest version, which should be released soon (6.18)."

Great news. Thanks for your investigation!
